ff/serving/metadata/dashboard/dashboard_metadata.go

```diff
@@ -213,23 +388,54 @@ func (m *MetadataServer) readFromSource(source *metadata.Source) (map[string]Sou
 		return nil, fetchError
 	}
 	for _, variant := range variants {
-		variantMap[variant.Variant()] = SourceVariantResource{
-			Created:      variant.Created(),
-			Description:  variant.Description(),
-			Name:         variant.Name(),
-			Type:         variant.Type(),
-			Variant:      variant.Name(),
-			Owner:        variant.Owner(),
-			Provider:     variant.Provider(),
-			Labels:       variant.Labels(),
-			Features:     variant.Features(),
-			TrainingSets: variant.TrainingSets(),
+
+		sourceResource := sourceShallowMap(variant)
+		if deepCopy {
+			errChan := make(chan *FetchError)
+			go func() {
+				f, err := m.nameVariantToFeatureList(variant.Features())
```

Comment:
Create a list of func() and loop through them and go run them. Then use the length of the slice when waiting for responses

Reply:
Nvm just use ErrGroup